### ISSUE TITLE
Query, Mutation, Subscription code are not generated from codegen (Android)

### ISSUE THREAD

**Issue by richardissuperman** [aws-amplify/amplify-android#582]
>**Note**: If your question is regarding the AWS Amplify Console service, please log it in the 
[official AWS Amplify Console forum](https://forums.aws.amazon.com/forum.jspa?forumID=314&start=0)

**Which Category is your question related to?**
Codegen

**Amplify CLI Version** 
4.20.0

**What AWS Services are you utilizing?**

AppSync, Amplify

**Provide additional details e.g. code snippets**

I have my ```schema.graphql``` file ready in my android project with my mutation and query and also data types(I didnt put the full file but just piece of it):

```
type Queue {
}

type Mutation {
	addQueue(input: AddQueueInput!): Queue
}

type Subscription {
	onAddQueue(customerId: ID!): Queue
		@aws_subscribe(mutations: ["addQueue"])
	onDeleteQueueEntity(queueId: ID!): String
		@aws_subscribe(mutations: ["deleteQueueEntity"])
}
```

After executing the **codegen** gradle task (in the terminal it shows codegen is successful), i can see only the model classes are there,  however , there are no mutation and query and subscription code.

I can see a empty Mutation.java/Query.java/Subscription.java file with nothing inside:
<img width="689" alt="Screen Shot 2020-06-06 at 4 51 23 PM" src="https://user-images.githubusercontent.com/5363585/83956738-fff13980-a815-11ea-9268-a82aa4a0b30b.png">

Noted im not using any directive in our schema.graphql file since we are not planning to use dynamodb or any other feature directive provides.

Thanks!

**Comment by jamesonwilliams** [comment]
@SwaySway It looks like the CLI is generating an empty Java document. That's a bug. The CLI should at least throw an error and refuse to create the document. All of that is happening before anything is run on a phone, no?

**Comment by SwaySway** [comment]
The CLI does not generate operations (queries and mutations) for non-model types. Is it still possible to use the Amplify API without non-model types? Ref pr: https://github.com/aws-amplify/amplify-android/pull/398

cc @yuth

**Comment by jamesonwilliams** [comment]
@richardissuperman @SwaySway -- the API and DataStore categories were designed with the `@model`s in mind. Support was added later so that an `@model` could contain non-`@model` types inside of it. @drochetti can provide more context on this feature.

If you _aren't_ using `@model`s _at all_, the API and DataStore categories might not be the best tool for the job. I'd suggest using the AWS AppSync SDK and working with the Amplify CLI team on code-gen.

**Comment by richardissuperman** [comment]
@jamesonwilliams we will definitely want to make use the great caching that Amplify provides, so  using @model directive is the right way to move forward for us,  I was confused since @SwaySway 
 mentioned that Amplify CLI works fine without directives.  

But looks like as long as client side define the model by using @model and @connection and perform an Amplify CLI Push, in app sync console it will automatically generates CRUD graphql schemas for the models. Is there a documentation on what is the naming convention for those generated CRUD graphql schemas? For example if i created a ```@model type Queue```, appsync will help generate 

```
type Mutation CreateQueue{

}
```

Basically our service team has designed a graphql schema and i want to see how client can have our own schema with @model directives and somehow match our service's schema, or we need to let service team to change their schema to match client side need.

**Comment by SwaySway** [comment]
@richardissuperman 
You can define what the name of the queries will be.
`@model` docs: https://docs.amplify.aws/cli/graphql-transformer/directives#model
